src/net_processing.cpp

```diff
@@ -1098,7 +1098,7 @@ void PeerManagerImpl::PushNodeVersion(CNode& pnode, int64_t nTime)
                            CAddress(CService(), addr.nServices);
     CAddress addrMe = CAddress(CService(), nLocalNodeServices);
 
-    const bool tx_relay = !m_ignore_incoming_txs && pnode.m_tx_relay != nullptr;
+    const bool tx_relay = !m_ignore_incoming_txs && pnode.m_tx_relay != nullptr && !pnode.IsFeelerConn();
```

Comment:
Doesn't this mean we won't request tx relay, but are still happy to process them?

Wouldn't it be better if this check here wasn't needed and `m_tx_relay` was never initialized? See also https://github.com/bitcoin/bitcoin/pull/21160#discussion_r693531183

Reply:
> nit: Doesn't this mean we won't request tx relay, but are (theoretically) still happy to process them?

No. For feeler connections, we disconnect as soon as we receive the `version` message, and we don't process any other message types before receiving `version`, so it's not possible for us to process any incoming transactions or announcements.

> Wouldn't it be better if this check here wasn't needed and m_tx_relay was never initialized?

I agree that it'd be better if `m_tx_relay` wasn't initialized until we receive the `version` message, and have implemented that in #22778. However, I don't think that's relevant here. For outbound peers, `PushNodeVersion` is sent _before_ receiving the `version` message, so this check can't be dependent on whether `m_tx_relay` has been initialized.